packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -298,67 +398,166 @@ abstract class Widget {
+/// information in the object itself and the [BuildContext] in which the widget
+/// is inflated. For compositions that can change dynamically, e.g. due to
+/// having an internal clock-driven state, or depending on some system state,
+/// consider using [StatefulWidget].
+///
+/// See also:
+///
+///  * [StatefulWidget]
 abstract class StatelessWidget extends Widget {
+  /// Initializes [key] for subclasses.
   const StatelessWidget({ Key key }) : super(key: key);
 
-  /// StatelessWidget always use [StatelessElement]s to represent
-  /// themselves in the Element tree.
+  /// Creates a [StatelessElement] to manage this widget's location in the tree.
+  ///
+  /// It is uncommon for subclasses to override this function.
   @override
   StatelessElement createElement() => new StatelessElement(this);
 
-  /// Returns another Widget out of which this StatelessWidget is built.
-  /// Typically that Widget will have been configured with further children,
-  /// such that really this function returns a tree of configuration.
+  /// Describes the part of the user interface represented by this widget.
   ///
-  /// The given build context object contains information about the location in
-  /// the tree at which this widget is being built. For example, the context
-  /// provides the set of inherited widgets for this location in the tree.
+  /// The framework calls this function when this widget is inserted into the
+  /// tree in a given [BuildContext] and when the dependencies of this widget
+  /// change (e.g., an [InheritedWidget] referenced by this widget changes).
+  ///
+  /// The framework replaces replaces the subtree below this widget with the
+  /// widget returned by this function, either by updating the existing subtree
+  /// or by removing the subtree and inflating a new subtree, depending on
+  /// whether the widget returned by this function [Widget.canUpdate] the root
+  /// of the existing subtree.
+  ///
+  /// Typically implementations return a newly created constellation of widget
+  /// that are configured with information from this widget's constructor and
+  /// from the given [BuildContext].
+  ///
+  /// The given [BuildContext] contains information about the location in the
+  /// tree at which this widget is being built. For example, the context
+  /// provides the set of inherited widgets for this location in the tree. A
+  /// given widget might be with multiple different [BuildContext] arguments
+  /// over time if the widget is moved around the tree or if the widget is
+  /// inserted into the tree in multiple places at once.
   @protected
   Widget build(BuildContext context);
 
   /// Trampoline to make the [build] closure library-accessible.
   WidgetBuilder get _build => build;
 }
 
-/// StatefulWidgets provide the configuration for
-/// [StatefulElement]s, which wrap [State]s, which hold mutable state
-/// and can dynamically and spontaneously ask to be rebuilt.
+/// A widget that has mutable state.
+///
```

Comment:
this would be a good place to define state. i forget what i had but my definition is in my talk if you have the slides for that